packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -526,6 +640,87 @@ class _RawAutocompleteState<T extends Object> extends State<RawAutocomplete<T>>
   }
 }
 
+/// Positions the options view.
+class _OptionsLayoutDelegate extends SingleChildLayoutDelegate {
+  _OptionsLayoutDelegate({
+    required this.fieldSize,
+    required this.fieldOffset,
+    required this.optionsViewOpenDirection,
+    required TextDirection? textDirection,
+  }) : textDirection = textDirection ?? TextDirection.ltr;
+
+  /// The size of the field in [RawAutocomplete.fieldViewBuilder].
+  final Size? fieldSize;
+
+  /// The position of the field in [RawAutocomplete.fieldViewBuilder].
+  final Offset? fieldOffset;
+
+   /// A direction in which to open the options view overlay.
+  final OptionsViewOpenDirection optionsViewOpenDirection;
+
+  /// The [TextDirection] of this part of the widget tree.
+  final TextDirection textDirection;
+
+  // Limits the child to the space above/below the field, with a minimum, and
+  // with the same maxWidth constraint as the field has.
+  @override
+  BoxConstraints getConstraintsForChild(BoxConstraints constraints) {
+    // shouldRelayout prevents this from being called if fieldOffset or
+    // fieldSize are invalid.
+    assert(fieldSize != null && fieldSize!.isFinite);
+    assert(fieldOffset != null && fieldOffset!.isFinite);
+
+    return constraints.loosen().copyWith(
+      // The field width may be zero if this is a split RawAutocomplete with no
+      // field of its own. In that case, don't change the constraints width.
+      maxWidth: fieldSize!.width == 0.0 ? constraints.maxWidth : fieldSize!.width,
+      maxHeight: switch (optionsViewOpenDirection) {
+        OptionsViewOpenDirection.down => max(
+          min(kMinInteractiveDimension, constraints.maxHeight),
+          constraints.maxHeight - fieldOffset!.dy - fieldSize!.height,
+        ),
+        OptionsViewOpenDirection.up => max(kMinInteractiveDimension, fieldOffset!.dy),
+      },
+    );
+  }
+
+  // Positions the child above/below the field and aligned with the left/right
+  // side based on text direction.
+  @override
+  Offset getPositionForChild(Size size, Size childSize) {
+    // shouldRelayout prevents this from being called if fieldOffset or
+    // fieldSize are invalid.
+    assert(fieldSize != null && fieldSize!.isFinite);
+    assert(fieldOffset != null && fieldOffset!.isFinite);
+
+    final double origin = size.height / 2 - fieldSize!.height / 2;
```

Comment:
I see. Does this fully solve that problem, though? What if the options list is tall, like more than half the height of the screen — will that cause the top of the list to extend up beyond the bounds of the parent?

Conversely, if the open direction is .down, it seems like this would cause a tall options list to extend down beyond the bounds of the parent, so that the items in the bottom half of the list couldn't be hit.

I think an alternate solution would be to have the `targetAnchor` and `followerAnchor` depend on the open direction, similar to how they do in main.

Reply:
> I see. Does this fully solve that problem, though? What if the options list is tall, like more than half the height of the screen — will that cause the top of the list to extend up beyond the bounds of the parent?

Yes, this is true. When I place the field at the top edge of the viewport and add a long list of options, latter options aren't hit-tested.

Reply:
This is fixed now! The follower is always anchored to the top left of the target, but in the `.up` open direction its bottom left is anchored while in the `.down` direction its top left is anchored.